core/src/main/java/com/sequenceiq/cloudbreak/repository/InstanceMetaDataRepository.java

```diff
@@ -46,6 +77,14 @@ public interface InstanceMetaDataRepository extends CrudRepository<InstanceMetaD
             "AND i.instanceStatus <> 'DELETED_BY_PROVIDER'")
     Set<InstanceMetaData> findNotTerminatedAndNotZombieForStack(@Param("stackId") Long stackId);
 
+    @Query("SELECT i.id FROM InstanceMetaData i " +
+            "WHERE i.instanceGroup.stack.id= :stackId " +
+            "AND i.instanceStatus <> 'TERMINATED' " +
```

Comment:
simplify: `not in ()`

Reply:
why is it simpler?

Reply:
I can changes here, but in the code we have lot of where clause like this

Reply:
🤷🏻